src/panels/lovelace/cards/energy/hui-energy-distribution-card.ts

```diff
@@ -154,55 +159,38 @@ class HuiEnergyDistrubutionCard
 
     let solarConsumption: number | null = null;
     if (hasSolarProduction) {
-      solarConsumption =
-        (totalSolarProduction || 0) -
-        (returnedToGrid || 0) -
-        (totalBatteryIn || 0);
+      solarConsumption = consumption.total.used_solar;
     }
-
     let batteryFromGrid: null | number = null;
     let batteryToGrid: null | number = null;
-    if (solarConsumption !== null && solarConsumption < 0) {
-      // What we returned to the grid and what went in to the battery is more than produced,
-      // so we have used grid energy to fill the battery
-      // or returned battery energy to the grid
-      if (hasBattery) {
-        batteryFromGrid = solarConsumption * -1;
-        if (batteryFromGrid > totalFromGrid) {
-          batteryToGrid = batteryFromGrid - totalFromGrid;
-          batteryFromGrid = totalFromGrid;
-        }
-      }
-      solarConsumption = 0;
+    if (hasBattery) {
+      batteryToGrid = consumption.total.battery_to_grid;
+      batteryFromGrid = consumption.total.grid_to_battery;
     }
 
     let solarToBattery: null | number = null;
     if (hasSolarProduction && hasBattery) {
-      if (!batteryToGrid) {
-        batteryToGrid = Math.max(
-          0,
-          (returnedToGrid || 0) -
-            (totalSolarProduction || 0) -
-            (totalBatteryIn || 0) -
-            (batteryFromGrid || 0)
+      solarToBattery = 0;
+      summedData.timestamps.forEach((t) => {
+        solarToBattery! += Math.max(
+          Math.min(
+            summedData.solar![t] || 0,
+            (summedData.to_battery![t] || 0) -
+              (consumption.grid_to_battery![t] || 0)
+          ),
+          0
         );
-      }
-      solarToBattery = totalBatteryIn! - (batteryFromGrid || 0);
-    } else if (!hasSolarProduction && hasBattery) {
-      batteryToGrid = returnedToGrid;
+      });
```

Comment:
Can we put`solarToBattery` in the unified calculations too?

Reply:
Added solar_to_battery and solar_to_grid